iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -1273,6 +1274,510 @@ public void testScanTableCaseInsensitive() throws IOException {
+            optional(2, "customer_id", Types.LongType.get()),
+            optional(3, "last_name", Types.StringType.get(), "This is last name"));
+
+    TestHelper.RecordsBuilder customersWithLastNameFirstBuilder =
+        TestHelper.RecordsBuilder.newInstance(customerSchemaLastNameFirst).add("Alice", 0L, "Brown")
+            .add("Bob", 1L, "Green").add("Trudy", 2L, "Pink");
+    List<Record> customersWithLastNameFirst = customersWithLastNameFirstBuilder.build();
+
+    // Run a 'select *' to check if the order of the column in the result has been changed.
+    List<Object[]> rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithLastNameFirst,
+        HiveIcebergTestUtils.valueForRow(customerSchemaLastNameFirst, rows), 1);
+
+    // Query the data with names and check if the result is the same as when the table was created.
+    rows = shell.executeStatement("SELECT customer_id, first_name, last_name FROM default.customers");
+    HiveIcebergTestUtils.validateData(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS,
+        HiveIcebergTestUtils.valueForRow(HiveIcebergStorageHandlerTestUtils.CUSTOMER_SCHEMA, rows), 0);
+
+    // Insert data from Hive to check if the last_name column has to be before the customer_id in the values list.
+    shell.executeStatement("INSERT INTO default.customers values ('Lily', 3L, 'Magenta')");
+
+    customersWithLastNameFirstBuilder.add("Lily", 3L, "Magenta");
+    customersWithLastNameFirst = customersWithLastNameFirstBuilder.build();
+    rows = shell.executeStatement("SELECT * FROM default.customers");
+    HiveIcebergTestUtils.validateData(customersWithLastNameFirst,
+        HiveIcebergTestUtils.valueForRow(customerSchemaLastNameFirst, rows), 1);
+  }
+
+  @Test
+  public void testUpdateColumnTypeInIcebergTable() throws IOException {
+    // Create an Iceberg table with int, float and decimal(2,1) types with some initial records
+    Schema schema = new Schema(optional(1, "id", Types.LongType.get()),
+        optional(2, "int_col", Types.IntegerType.get(), "This is an integer type"),
+        optional(3, "float_col", Types.FloatType.get(), "This is a float type"),
+        optional(4, "decimal_col", Types.DecimalType.of(2, 1), "This is a decimal type"));
+
+    List<Record> records = TestHelper.RecordsBuilder.newInstance(schema).add(0L, 35, 22F, BigDecimal.valueOf(13L, 1))
+        .add(1L, 223344, 555.22F, BigDecimal.valueOf(22L, 1)).add(2L, -234, -342F, BigDecimal.valueOf(-12L, 1)).build();
+
+    Table icebergTable = testTables.createTable(shell, "types_table", schema, fileFormat, records);
+
+    Schema schemaForResultSet =
+        new Schema(optional(1, "id", Types.LongType.get()), optional(2, "int_col", Types.IntegerType.get()),
+            optional(3, "float_col", Types.DoubleType.get()), optional(4, "decimal_col", Types.StringType.get()));
```

Comment:
Shouldn't this be `Types.FloatType.get()` and `Types.DecimalType.of(2, 1)`?

Reply:
Actually in the result set a float column is returned as double and a decimal is returned as string, even though Hive has the columns with the right types. I guess this conversation happen when fetching the result set after calling the select through the shell. I didn't dig deeper but I can do it if you want.

Reply:
Ah I see. It's okay then. Maybe just add a comment, explaining that we have to use these types here because of the fetch task type conversion

Reply:
Added a comment about that.